tests/integration/main.rs

```diff
@@ -363,3 +363,318 @@ fn orc_split_elim_new() {
+
+    // Read all batches
+    let batches: Result<Vec<_>, _> = reader.collect();
+    assert!(batches.is_ok());
+    let batches = batches.unwrap();
+    assert!(!batches.is_empty());
+
+    // Verify total row count (should be 1000 rows)
+    let total_rows: usize = batches.iter().map(|b| b.num_rows()).sum();
+    assert_eq!(
+        total_rows, 1000,
+        "Expected 1000 rows in bloom_filter_test.orc"
+    );
+
+    // Verify schema has expected columns
+    let schema = batches[0].schema();
+    assert_eq!(schema.fields().len(), 4, "Expected 4 columns");
+    assert_eq!(schema.field(0).name(), "id");
+    assert_eq!(schema.field(1).name(), "name");
+    assert_eq!(schema.field(2).name(), "age");
+    assert_eq!(schema.field(3).name(), "email");
+}
+
+#[test]
+fn test_bloom_filter_equality_query_existing_value() {
+    // Test equality query with a value that exists in the file
+    // Bloom Filter should allow reading the row group, but may return the entire row group
+    // rather than just the matching rows (predicate pushdown is at row group level)
+    let path = format!(
+        "{}/tests/integration/data/bloom_filter_test.orc",
+        env!("CARGO_MANIFEST_DIR"),
+    );
+    let f = File::open(path).unwrap();
+
+    // Query for id = 500 (exists in the file)
+    let predicate = Predicate::eq("id", PredicateValue::Int32(Some(500)));
+    let reader = ArrowReaderBuilder::try_new(f)
+        .unwrap()
+        .with_predicate(predicate)
+        .build();
+
+    let batches: Result<Vec<_>, _> = reader.collect();
+    assert!(batches.is_ok());
+    let batches = batches.unwrap();
+
+    // Should return at least some rows (Bloom Filter allows reading the row group)
+    let total_rows: usize = batches.iter().map(|b| b.num_rows()).sum();
+    assert!(
+        total_rows > 0,
+        "Should return rows when Bloom Filter indicates value might exist"
+    );
+
+    // Verify that id=500 is in the returned data (since it exists in the file)
+    let mut found_500 = false;
+    for batch in &batches {
+        let id_array = batch
+            .column(0)
+            .as_primitive_opt::<arrow::datatypes::Int32Type>()
+            .unwrap();
+        for i in 0..id_array.len() {
+            if !id_array.is_null(i) && id_array.value(i) == 500 {
+                found_500 = true;
+                break;
+            }
+        }
+        if found_500 {
+            break;
+        }
+    }
+    assert!(found_500, "Should find id=500 in the returned data");
+}
+
+#[test]
+fn test_bloom_filter_equality_query_nonexistent_value() {
+    // Test equality query with a value that doesn't exist
+    // Bloom Filter should help skip row groups that don't contain the value
+    let path = format!(
+        "{}/tests/integration/data/bloom_filter_test.orc",
+        env!("CARGO_MANIFEST_DIR"),
+    );
+    let f = File::open(path).unwrap();
+
+    // Query for id = 9999 (doesn't exist, ids are 1-1000)
```

Comment:
This is not sufficient to check the row group was eliminated by BF because statistics would also eliminate it.